src/platform/cmx/boot_comp.c

```diff
@@ -0,0 +1,424 @@
+	int ret;
+	u8_t *ptes;
+	unsigned int nptes = 0, i;
+	struct cap_pgtbl *pgd_cap, *uvmpgd_cap, *shpgd_cap;
+	pgtbl_t pgtbl, uvmpgtbl, shpgtbl;
+
+	/* Setup the untyped 4k memory segments first */
+	uvmpgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, uvmpgdcap);
+	if (!uvmpgd_cap || !CAP_TYPECHK(uvmpgd_cap, CAP_PGTBL)) assert(0);
+	uvmpgtbl = (pgtbl_t)&(uvmpgd_cap->pgtbl);
+
+	uvmpgtbl->data.pgt[0]=(addr+0x8000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[1]=(addr+0x8000+0x1000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[2]=(addr+0x8000+0x2000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[3]=(addr+0x8000+0x3000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[4]=(addr+0x8000+0x4000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[5]=(addr+0x8000+0x5000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[6]=(addr+0x8000+0x6000)|CMX_PGTBL_COSFRAME;
+	uvmpgtbl->data.pgt[7]=(addr+0x8000+0x7000)|CMX_PGTBL_COSFRAME;
+
+	/* Setup the freertos component's page table then */
+	pgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, pgdcap);
+	if (!pgd_cap || !CAP_TYPECHK(pgd_cap, CAP_PGTBL)) assert(0);
+	pgtbl = (pgtbl_t)&(pgd_cap->pgtbl);
+
+	/* Add kernel data regions - they shall never be retyped into anything else */
+	pgtbl->data.pgt[0]=addr|CMX_PGTBL_USER_DEF;
+	/* pgtbl->data.pgt[1]=0; something cons'ed into here already */
+
+	/* Retype the frt component's range to user memory */
+	for(unsigned long i=addr;i<addr+0x8000;i+=RETYPE_MEM_SIZE)
+		assert(retypetbl_retype2user((void*)i)==0);
+
+	/* Manually construct the initial MPU settings. After this, all the settings must be constructed by the update function.
+	 * here, we only grant access to the first MPU regions. All other code regions, by default, are enabled; and the background
+	 * region is also enabled. These are done by default */
+	chal_mpu_update(pgtbl, 1);
+
+	/* Setup the untyped 4k memory segments first */
+	shpgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, shpgdcap);
+	if (!shpgd_cap || !CAP_TYPECHK(shpgd_cap, CAP_PGTBL)) assert(0);
+	shpgtbl = (pgtbl_t)&(shpgd_cap->pgtbl);
+
+	shpgtbl->data.pgt[0]=0;
+	shpgtbl->data.pgt[1]=(0x20029000)|CMX_PGTBL_USER_DEF;
+	shpgtbl->data.pgt[2]=0;
+	shpgtbl->data.pgt[3]=0;
+	shpgtbl->data.pgt[4]=0;
+	shpgtbl->data.pgt[5]=0;
+	shpgtbl->data.pgt[6]=0;
+	shpgtbl->data.pgt[7]=0;
+	pgtbl=(pgtbl_t)&(shpgd_cap->pgtbl);
+	chal_mpu_update(pgtbl, 1);
+
+	return 0;
+}
+
+int boot_pgtbl_mappings_add(struct captbl *ct, capid_t pgdcap, capid_t uvm4k0pgdcap, capid_t uvm4k1pgdcap)
+{
+	u32_t count;
+	struct pgtbl* MPU_meta;
+	int ret;
+	u8_t *ptes;
+	unsigned int nptes = 0, i;
+	struct cap_pgtbl *pgd_cap, *uvm4k0pgd_cap, *uvm4k1pgd_cap;
+	pgtbl_t pgtbl, uvm4k0pgtbl, uvm4k1pgtbl;
+
+	/* Setup the untyped 4k memory segments first */
+	uvm4k0pgd_cap = (struct cap_pgtbl*)captbl_lkup(ct, uvm4k0pgdcap);
+	if (!uvm4k0pgd_cap || !CAP_TYPECHK(uvm4k0pgd_cap, CAP_PGTBL)) assert(0);
+	uvm4k0pgtbl = (pgtbl_t)&(uvm4k0pgd_cap->pgtbl);
+
+	uvm4k0pgtbl->data.pgt[0]=(COS_CMX_SECOND_PAGE+0x0000)|CMX_PGTBL_COSFRAME;
```

Comment:
You should go through your changes to fix the spacing to follow CSG (Composite Style Guide).